PROTO_tests/tests/numeric_test.py

```diff
@@ -206,13 +205,242 @@ def test_sin(self, num_type):
 
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_tan(self, num_type):
+        if num_type == ak.float64:
+            na = np.linspace(0, 10)
+        else:
+            na = np.arange(0, 10).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.tan(na), ak.tan(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.tan(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arcsin(self, num_type):
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arcsin(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arccos(self, num_type):
+        print("mytest")
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arccos(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arctan(self, num_type):
+        if num_type == ak.float64:
+            na = np.linspace(0, 10)
+        else:
+            na = np.arange(0, 10).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray())
+        with pytest.raises(TypeError):
+            ak.arctan(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: infinities
+        if num_type == ak.float64:
+            na = np.array([np.inf, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    @pytest.mark.parametrize("denom_type", NO_BOOL)
+    def test_arctan2(self, num_type, denom_type):
+        if num_type == ak.float64:
+            na_num = np.linspace(0, 1, 10)
+            pda_num = ak.array(na_num, dtype=num_type)
+        else:
+            na_num = np.arange(10, 0, -1).astype(num_type)
+            pda_num = ak.array(na_num, dtype=num_type)
+        if denom_type == ak.float64:
+            na_denom = np.linspace(0, 10, 10)
+            pda_denom = ak.array(na_denom, dtype=denom_type)
+        else:
+            na_denom = np.arange(0, 10).astype(denom_type)
+            pda_denom = ak.array(na_denom, dtype=denom_type)
```

Comment:
`pda_num/denom` is the same in all these cases, so we should be able to pull it out.
```python
if num_type == ak.float64:
    na_num = np.linspace(0, 1, 10)
else:
    na_num = np.arange(10, 0, -1).astype(num_type)
if denom_type == ak.float64:
    na_denom = np.linspace(0, 10, 10)
else:
    na_denom = np.arange(0, 10).astype(denom_type)
pda_num = ak.array(na_num, dtype=num_type)
pda_denom = ak.array(na_denom, dtype=denom_type)
```

We could also turn the others into ternaries, but im not sure if that's easier to read or not
```python
na_num = np.linspace(0, 1, 10) if num_type == ak.float64 else np.arange(10, 0, -1).astype(num_type)
na_denom = np.linspace(0, 10, 10) if denom_type == ak.float64 else np.arange(0, 10).astype(denom_type)
pda_num = ak.array(na_num, dtype=num_type)
pda_denom = ak.array(na_denom, dtype=denom_type)
```

Reply:
Updated